### PR TITLE
swift 2.2-SNAPSHOT-2015-12-18-a (new formula)

### PR DESCRIPTION
**Pull request by mikemcquaid**
Bit of a pain in the arse to get this building so it's possibly not including everything we'd want to include (e.g. the Swift package manager). However, probably better to get _something_ in core to give people a block to build upon.

### CODE REVIEW

**Comment by mikemcquaid** [comment]
Failing on 10.9 but I'll just add the relevant `depends_on` for the Xcode version when I pull.

File: Library/Formula/swift.rb [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+        "--skip-test-xctest",
+        "--skip-test-foundation",
+        "--skip-test-validation",
+        "--skip-test-optimized",
+        "--skip-build-lldb",
+        "--skip-build-llbuild",
+        "--skip-build-swiftpm",
+        "--skip-build-xctest",
+        "--skip-build-foundation",
+        "--build-args=-j8",
```
Comment (bfontaine) [inline Library/Formula/swift.rb]: Shouldn’t that depend on something like `ENV.make_jobs`?

Comment (mikemcquaid) [inline Library/Formula/swift.rb]: Yeh, whoops, hardcoded it locally. Again, will fix when I pull to avoid another (long) build.

**Comment by mikemcquaid** [comment]
Pulled with the various comments fixed.

File: Library/Formula/swift.rb [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+    resource "llvm" do
+      url "https://github.com/apple/swift-llvm.git", :branch => "stable"
+    end
+  end
+
+  keg_only :provided_by_osx, "Apple's CLT package contains Swift."
+
+  depends_on "cmake" => :build
+  depends_on "ninja" => :build
+  depends_on :xcode => :build
```
Comment (mistydemeo) [inline Library/Formula/swift.rb]: Couldn't get CLT-only build working?

Comment (mikemcquaid) [inline Library/Formula/swift.rb]: Unfortunately not. I'm sure it's possible though but after building this ~20 times I wanted to just get _something_ in :grinning:

File: Library/Formula/swift.rb [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+
+  def install
+    workspace = buildpath.parent
+    build = workspace/"build"
+
+    ln_sf buildpath, "#{workspace}/swift"
+    resources.each { |r| r.stage("#{workspace}/#{r.name}") }
```
Comment (xu-cheng) [inline Library/Formula/swift.rb]: Wouldn't it stage files outside of `buildpath`? (Noted `workspace` is outside of `buildpath`) Will files get cleanup probably(especially when building error or interrupt)?

Comment (mikemcquaid) [inline Library/Formula/swift.rb]: Outside the `buildpath` but inside the temporary directory that's created/deleted.

Comment (xu-cheng) [inline Library/Formula/swift.rb]: :cool: (Though, we may want to avoid such use case as it's kinda confusing and relies on our implementation detail)

Comment (mikemcquaid) [inline Library/Formula/swift.rb]: Agreed.

**Pull request closed without merging**